daemon/container.go

```diff
@@ -1275,19 +1277,31 @@ func (container *Container) initializeNetworking() error {
 	return container.buildHostnameAndHostsFiles(container.NetworkSettings.IPAddress)
 }
 
-// Make sure the config is compatible with the current kernel
-func (container *Container) verifyDaemonSettings() {
-	if container.hostConfig.Memory > 0 && !container.daemon.sysInfo.MemoryLimit {
-		logrus.Warnf("Your kernel does not support memory limit capabilities. Limitation discarded.")
-		container.hostConfig.Memory = 0
+// Make sure the config is compatible with the current kernel and daemon
+func (container *Container) verifyDaemonSettings() error {
+	hostConfig := container.hostConfig
+	daemon := container.daemon
+
+	if hostConfig.LxcConf.Len() > 0 && !strings.Contains(daemon.ExecutionDriver().Name(), "lxc") {
```

Comment:
@hqhq, when starting containers, it really need check settings. But ContainerStart() already calls verifyHostConfig do it before calling container.Start().
So, there is not necessary to do double check